### DISCUSSION TITLE
Can't logout with SSR

### DISCUSSION THREAD

**Discussion by jaywei1234** [pocketbase/pocketbase#2229] [Q&A]
I was tryin to use the pb.authStore.clear() to logout, but the problem is I can't delete the pb_auth cookie to finsih the logout, what is the problem?

I was doing something like this.
![image](https://user-images.githubusercontent.com/56056463/230361873-4246b932-1545-4072-ad10-e8e2a768704e.png)

I was doin the logout function in a component page, is it the problem?

**Comment by ganigeorgiev** [comment]
I don't understand the above code. Why do you need a `onChange` function in the first place? The way it is defined it will get triggered by events **after** the function registration.

If you just want to update the cookie after `pb.authStore.clear()`, you can either move it after the `onChange` call OR just call the `document.cookie` right after the clear call (aka. without the `onChange` wrapping).

**Reply by jaywei1234** [reply]
I thought it can just update the pb_auth cookie for clearing the cookie

Because pb.authStore.clear() just clear the localStorage but not the cookie

**Reply by ganigeorgiev** [reply]
But again, if this is a one off client instance, why do you need a `onChange` call at all? The following should be enough:

```js
pb.authStore.clear()

document.cookie = pb.authStore.exportToCookie({ httpOnly: false })
```

**Reply by jaywei1234** [reply]
got it, thank you!!

**Reply by jaywei1234** [reply]
It still not clear the pb_auth cookies with the above code

**Reply by ganigeorgiev** [reply]
The above will set a browser `pb_auth` cookie with expired duration.
I don't understand what you are trying to do, nor how the frontend app interact with the PocketBase server. Please elaborate with a little more detailed code sample and information about your setup.

**Reply by jaywei1234** [reply]
Is the above code should run inside getServerSideProps? Because I just put the above code in a function called logout, and after the user click on the logout button, it calls the function.

**Reply by jaywei1234** [reply]
For your reference
![image](https://user-images.githubusercontent.com/56056463/230431133-9a55815c-0663-4144-84ba-d7a3670b8dfd.png)
It is the ssr code for me to check auth on every page
